chainguard-dev/edu
# Refactor rumble to generate all csv and json files using insights data

## Type of change
rumble improvement

### What should this PR do?
This PR cuts rumble over to the insights database. It switches the nightly CSV and JSON generation for vulnerability comparison pages to use a single Go binary, instead of using 3 different GitHub actions.

### Why are we making this change?
Insights is the authoritative source for image scans now, and the GitHub actions rely on using gcloud commands, which were failing and leaking auth details in unexpected ways.

### What are the acceptance criteria? 
The actions in this PR should run and generate files.

### How should this PR be tested?
This PR should trigger the actions. But running the tool locally should work too:

```
go run main.go legacy-csv --project prod-images-c6e5  --db insights_ds --gcs-project chainguard-academy --bucket chainguard-academy 
go run main.go vulns --project prod-images-c6e5  --db insights_ds --gcs-project chainguard-academy --bucket chainguard-academy
go run main.go image-csv --project prod-images-c6e5  --db insights_ds --gcs-project chainguard-academy --bucket chainguard-academy --rumble-json-path ../../data/rumble.json
